### ISSUE TITLE
Compile error when npm installing srs

### ISSUE THREAD

**Issue by jecor** [mapbox/node-srs#60]
From dependency of npm install millstone:

  CXX(target) Release/obj.target/osr/deps/osr/src/ogr_srs_validate.o
  CXX(target) Release/obj.target/osr/deps/osr/src/ogr_srs_xml.o
  CXX(target) Release/obj.target/osr/deps/osr/src/ogr_srsnode.o
../deps/osr/src/ogr_srsnode.cpp:159:9: warning: 'this' pointer cannot be null in
      well-defined C++ code; comparison may be assumed to always evaluate to
      false [-Wtautological-undefined-compare]
    if( this == NULL )
        ^~~~    ~~~~
1 warning generated.
  CXX(target) Release/obj.target/osr/deps/osr/src/ogrct.o
../deps/osr/src/ogrct.cpp:772:9: warning: using integer absolute value function
      'abs' when argument is of floating point type [-Wabsolute-value]
        abs(dfSourceToRadians * dfTargetFromRadians - 1.0) < 1e-10 )
        ^
../deps/osr/src/ogrct.cpp:772:9: note: use function 'std::abs' instead
        abs(dfSourceToRadians * dfTargetFromRadians - 1.0) < 1e-10 )
        ^~~
        std::abs
../deps/osr/src/ogrct.cpp:772:9: note: include the header <cmath> or explicitly
      provide a declaration for 'std::abs'
1 warning generated.
  CXX(target) Release/obj.target/osr/deps/osr/src/ogrspatialreference.o
  LIBTOOL-STATIC Release/osr.a
  CXX(target) Release/obj.target/srs/src/srs.o
In file included from ../src/srs.cc:6:
../node_modules/nan/nan.h:261:25: error: redefinition of '_NanEnsureLocal'
NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
                        ^
../node_modules/nan/nan.h:256:25: note: previous definition is here
NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
                        ^
../node_modules/nan/nan.h:661:13: error: no member named 'smalloc' in namespace
      'node'
    , node::smalloc::FreeCallback callback
      ~~~~~~^
../node_modules/nan/nan.h:672:12: error: no matching function for call to 'New'
    return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
           ^~~~~~~~~~~~~~~~~
/var/root/.node-gyp/4.2.6/include/node/node_buffer.h:31:40: note: candidate
      function not viable: no known conversion from 'uint32_t'
      (aka 'unsigned int') to 'enum encoding' for 3rd argument
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/var/root/.node-gyp/4.2.6/include/node/node_buffer.h:43:40: note: candidate
      function not viable: 2nd argument ('const char *') would lose const
      qualifier
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/var/root/.node-gyp/4.2.6/include/node/node_buffer.h:28:40: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
                                       ^
/var/root/.node-gyp/4.2.6/include/node/node_buffer.h:36:40: note: candidate
      function not viable: requires 5 arguments, but 3 were provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
In file included from ../src/srs.cc:6:
../node_modules/nan/nan.h:676:12: error: no viable conversion from
      'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object>'
    return node::Buffer::New(v8::Isolate::GetCurrent(), size);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/root/.node-gyp/4.2.6/include/node/v8.h:210:7: note: candidate constructor
      (the implicit copy constructor) not viable: no known conversion from
      'v8::MaybeLocal<v8::Object>' to 'const v8::Local<v8::Object> &' for 1st
      argument
class Local {
      ^
/var/root/.node-gyp/4.2.6/include/node/v8.h:210:7: note: candidate constructor
      (the implicit move constructor) not viable: no known conversion from
      'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object> &&' for 1st
      argument
class Local {
      ^
/var/root/.node-gyp/4.2.6/include/node/v8.h:214:13: note: candidate template
      ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local(Local<S> that)
            ^
/var/root/.node-gyp/4.2.6/include/node/v8.h:326:13: note: candidate template
      ignored: could not match 'S *' against 'v8::MaybeLocal<v8::Object>'
  V8_INLINE Local(S* that)
            ^
In file included from ../src/srs.cc:6:
../node_modules/nan/nan.h:683:26: error: no member named 'Use' in namespace
      'node::Buffer'
    return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
           ~~~~~~~~~~~~~~^
In file included from ../src/srs.cc:2:
/var/root/.node-gyp/4.2.6/include/node/v8.h:221:5: error: assigning to
      'v8::Primitive *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/var/root/.node-gyp/4.2.6/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:414:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Primitive>::Local<v8::Value>' requested here
    return NanEscapeScope(NanNew(v8::Undefined(v8::Isolate::GetCurrent())));
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/srs.cc:2:
/var/root/.node-gyp/4.2.6/include/node/v8.h:221:5: error: assigning to
      'v8::Boolean *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/var/root/.node-gyp/4.2.6/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:424:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Boolean>::Local<v8::Value>' requested here
    return NanEscapeScope(NanNew(v8::True(v8::Isolate::GetCurrent())));
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/srs.cc:2:
/var/root/.node-gyp/4.2.6/include/node/v8.h:221:5: error: assigning to
      'v8::Function *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/var/root/.node-gyp/4.2.6/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1514:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Function>::Local<v8::Value>' requested here
    return NanEscapeScope(NanNew(handle)->Get(kCallbackIndex)
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/srs.cc:2:
/var/root/.node-gyp/4.2.6/include/node/v8.h:221:5: error: assigning to
      'v8::Object *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/var/root/.node-gyp/4.2.6/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1632:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Object>::Local<v8::Value>' requested here
    return NanEscapeScope(handle->Get(NanNew(key)).As<v8::Object>());
           ^
../node_modules/nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
9 errors generated.
make: *** [Release/obj.target/srs/src/srs.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/opt/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/opt/local/bin/node" "/opt/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/lib/binding/node-v46-darwin-x64/srs.node" "--module_name=srs" "--module_path=/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/lib/binding/node-v46-darwin-x64"
gyp ERR! cwd /Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/opt/local/bin/node /opt/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/lib/binding/node-v46-darwin-x64/srs.node --module_name=srs --module_path=/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/lib/binding/node-v46-darwin-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:821:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
node-pre-gyp ERR! System Darwin 14.5.0
node-pre-gyp ERR! command "/opt/local/bin/node" "/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs
node-pre-gyp ERR! node -v v4.2.6
node-pre-gyp ERR! node-pre-gyp -v v0.6.9
node-pre-gyp ERR! not ok 
Failed to execute '/opt/local/bin/node /opt/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/lib/binding/node-v46-darwin-x64/srs.node --module_name=srs --module_path=/Users/jerome/Documents/MapBox/project/TopoAuvergne/node_modules/millstone/node_modules/srs/lib/binding/node-v46-darwin-x64' (1)
npm ERR! Darwin 14.5.0
npm ERR! argv "/opt/local/bin/node" "/opt/local/bin/npm" "install" "millstone"
npm ERR! node v4.2.6
npm ERR! npm  v2.14.17
npm ERR! code ELIFECYCLE

npm ERR! srs@0.4.9 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the srs@0.4.9 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the srs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs srs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR! 
npm ERR!     npm owner ls srs
npm ERR! There is likely additional logging output above.

My clang:

clang --version
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin14.5.0
Thread model: posix

Any help would be appreciated :-)

**Comment by springmeyer** [comment]
You are using an old node-srs release. Upgrade to `v1.1.0` and that then uses https://github.com/naturalatlas/node-gdal and you won't hit this problem anymore.

**Comment by jecor** [comment]
Thanks for your quick help! Indeed, version 1.1.0 installs like a charm... except than when I try to install again millstone... it still requires an outdated version of srs :-(

Sorry for the stupid question (I am very new to node), but how to change this?

**Comment by springmeyer** [comment]
@jecor done in https://github.com/mapbox/millstone/commit/4c3a679641bec2494eec84616f56b63c13ff069e. So you can install millstone from master branch (not sure when I'll have time for a release). Does that work?

**Comment by springmeyer** [comment]
nevermind, just published millstone@0.6.17

**Comment by jecor** [comment]
Thanks a lot! millstone now installing like a charm as well.
